src/fixes/incorrect_delimiter.rs

```diff
@@ -0,0 +1,100 @@
+use super::Fix;
+use crate::common::*;
+
+pub(crate) struct IncorrectDelimiterFixer<'a> {
+    name: &'a str,
+}
+
+impl Default for IncorrectDelimiterFixer<'_> {
+    fn default() -> Self {
+        Self {
+            name: "IncorrectDelimiter",
+        }
+    }
+}
+
+impl Fix for IncorrectDelimiterFixer<'_> {
+    fn name(&self) -> &str {
+        self.name
+    }
+
+    fn fix_line(&self, line: &mut LineEntry) -> Option<()> {
+        let key: String = line.get_key()?;
+        let fixed_key: String = key
```

Comment:
Now I thought that we should not replace characters if the string begins or ends with them (thanks #254).
We can only call a character within keys a delimiter.
The rest of the cases are handled by other checks.

Reply:
I'm sorry, but I don't fully understand what you mean.
Could you give an example?

Reply:
> I'm sorry, but I don't fully understand what you mean.
> Could you give an example?

For example: 
`-BAR=FOO`
I think should be fixed by LeadingCharacter
and `FOO =BAR` should be fixed by `SpaceCharacter`
It's not a delimiter, because in second case, we will get: `FOO_=BAR` instead of `FOO=BAR`.

This can be influenced by the order, but given that checks can be skipped, I think this is not correct.

Reply:
My mistake about `FOO =BAR` case, sorry. 
I did not notice `trim`.

Most likely I'm wrong, but for some reason this question does not let me go to the end

Reply:
How would you implement this fixer?

Reply:
I think it's dependent from [another PR](https://github.com/dotenv-linter/dotenv-linter/pull/259#discussion_r470599906).
Solution must match check logic :thinking:

Reply:
Sure, we can wait until the other PR is merged and then have the fixer match the new check logic.

Reply:
FYI - see #267 for the PR updating the logic

Reply:
Sounds good, I will wait until that PR is merged before rebasing and updating.

Reply:
@gillespiecd #267 was merged. Please, update your PR from master :relaxed:

Reply:
Updated, thanks for the reminder 🚀